rtic-sync/src/signal.rs

```diff
@@ -0,0 +1,119 @@
+//! A "latest only" value store with unlimited writers and async waiting.
+
+use core::{
+    cell::UnsafeCell,
+    future::poll_fn,
+    sync::atomic::{fence, Ordering},
+    task::Poll,
+};
+use rtic_common::waker_registration::CriticalSectionWakerRegistration;
+
+/// Basically an Option but for indicating
+/// whether the store has been set or not
+#[derive(Clone, Copy)]
+enum Store<T> {
+    Set(T),
+    Unset,
+}
+
+/// A "latest only" value store with unlimited writers and async waiting.
+pub struct Signal<T: Copy> {
+    waker: CriticalSectionWakerRegistration,
+    store: UnsafeCell<Store<T>>,
+}
+
+unsafe impl<T: Copy> Send for Signal<T> {}
+unsafe impl<T: Copy> Sync for Signal<T> {}
+
+impl<T: Copy> Signal<T> {
+    /// Create a new signal.
+    pub const fn new() -> Self {
+        Self {
+            waker: CriticalSectionWakerRegistration::new(),
+            store: UnsafeCell::new(Store::Unset),
+        }
+    }
+
+    /// Split the signal into a writer and reader.
+    pub fn split(&'static self) -> (SignalWriter<T>, SignalReader<T>) {
```

Comment:
Is there a reason to force `'static` and not let the lifetime system figure it out?

Reply:
Of course, I've updated it to be generic.

Reply:
:rocket: